pkg/collector/corechecks/cluster/topologycollectors/container_correlator_test.go

```diff
@@ -1,43 +1,223 @@
-	go func() {
-		for _, cnc := range []*ContainerToNodeCorrelation{
-			{
-				NodeName: "test-node-3",
-				MappingFunction: func(nodeIdentifier string) (components []*topology.Component, relations []*topology.Relation) {
+import (
+	"fmt"
+	"github.com/StackVista/stackstate-agent/pkg/topology"
+	"github.com/StackVista/stackstate-agent/pkg/util/kubernetes/apiserver"
+	"github.com/stretchr/testify/assert"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"testing"
+	"time"
+)
+
+var startedAtTime = metav1.NewTime(time.Now())
+
+func TestContainerCollector(t *testing.T) {
+	componentChannel := make(chan *topology.Component)
+	defer close(componentChannel)
+	relationChannel := make(chan *topology.Relation)
+	defer close(relationChannel)
 
-					containerExternalID := ic.buildContainerExternalID("test-pod", "test-container")
-					podExternalID := ic.buildPodExternalID("test-pod")
+	nodeIdentifierCorrelationChannel := make(chan *NodeIdentifierCorrelation)
+	containerCorrelationChannel := make(chan *ContainerCorrelation)
 
-					components = append(components, &topology.Component{
-						ExternalID: containerExternalID,
-						Type: topology.Type{ Name: "container"},
+	cc := NewContainerCorrelator(componentChannel, relationChannel, nodeIdentifierCorrelationChannel,
+		containerCorrelationChannel, NewTestCommonClusterCorrelator(MockContainerAPICollectorClient{}))
+	expectedCollectorName := "Container Correlator"
+
+	populateData(nodeIdentifierCorrelationChannel, containerCorrelationChannel)
+
+	RunCorrelatorTest(t, cc, expectedCollectorName)
+
+	for _, tc := range []struct {
+		testCase   string
+		assertions []func()
+	}{
+		{
+			testCase: "Test Container 1",
+			assertions: []func(){
+				func() {
+					component := <-componentChannel
+					expectedComponent := &topology.Component{
+						ExternalID: "urn:kubernetes:/test-cluster-name:namespace-1:pod/Pod-Name-1:container/container-1",
+						Type:       topology.Type{Name: "container"},
+						Data: topology.Data{
+							"docker":       map[string]interface{}{"containerId": "containerID-1", "image": "image-1"},
+							"identifiers":  []string{"urn:container:/nodeID-1:containerID-1"},
+							"name":         "container-1",
+							"pod":          "Pod-Name-1",
+							"podIP":        "10.0.1.1",
+							"podPhase":     "Running",
+							"restartCount": int32(1),
+							"tags":         map[string]string{"cluster-name": "test-cluster-name", "namespace": "namespace-1"},
+						},
+					}
+					assert.EqualValues(t, expectedComponent, component)
+				},
+				func() {
+					relation := <-relationChannel
+					expectedRelation := &topology.Relation{
+						ExternalID: "Pod-ExternalID-1->urn:kubernetes:/test-cluster-name:namespace-1:pod/Pod-Name-1:container/container-1",
+						SourceID:   "Pod-ExternalID-1",
+						TargetID:   "urn:kubernetes:/test-cluster-name:namespace-1:pod/Pod-Name-1:container/container-1",
+						Type:       topology.Type{Name: "encloses"},
```

Comment:
i'd rather say "pod _has_ container"

Reply:
I think I would say the same, but I didn't change that part. It's just testing code that was already there. I'm not sure what this might also affect. Probably it would also break some other tests.